content/blog/query-kubernetes/index.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Introducing CloudQuery
+authors: ["joe-duffy"]
+tags: ["Pulumi-News", "Kubernetes"]
+meta_desc: "Today we announced Pulumi CloudQuery, an SDK for programmatically querying cloud resoures."
+date: "2019-11-14"
+meta_image: "pulumi-crosswalk-k8s.png"
+---
+
+Today, as part of our Crosswalk for Kubernetes announcement[link], we're introducing CloudQuery for
+Kubernetes.
+
+CloudQuery is a relational TypeScript SDK for querying live Kubernetes resources. It makes it easy
+to answer questions that would normally be answered by hand-scripting calls to `kubectl`. Questions
+like:
+
+* How many distinct versions of MySQL are running in my cluster?
+* Which Pods are scheduled on nodes with high memory pressure?
+* Which Pods are publicly exposed to the internet via a load-balanced Service?
+
+CloudQuery supports both batch and streaming query modes. Batch queries run on a finite set of
+resources. But streaming queries operate on an unbounded stream of resource updates, making it easy
+to watch Kubernetes resources and to take action when specific events occur. For example:
+
+* Post a slack notification when we run out of disk space.
+* Page someone when a rollout fails.
+
+In this post, we'll look at both batch and streaming query modes, showing examples of each. The SDK
+repository also contains a set of example queries, including instructions for how to run them.
```

Comment:
SDK repository -> [Pulumi Query for Kubernetes](https://github.com/pulumi/pulumi-query-kubernetes) repository